server/api/crud/artifacts.py

```diff
@@ -165,3 +171,15 @@ def _transform_artifact_struct_to_legacy_format(artifact):
         for key, value in artifact[section].items():
             legacy_artifact[key] = value
     return legacy_artifact
+
+
+def _resolve_artifact_size(data, auth_info):
```

Comment:
move this function to be part of artifact class and use `self._auto_info`
would be much easier

Reply:
do you mean use self._resolve_artifact_size ?

Reply:
this function is not part of the class that uses it. so better move it to be part of the class uses it. that way, you will have direct access to the auth_info and you wont need to get it as a parameter